rs/replicated_state/src/metadata_state.rs

```diff
@@ -62,7 +64,7 @@ pub type StreamMap = BTreeMap<SubnetId, Stream>;
 
 /// Replicated system metadata.  Used primarily for inter-canister messaging and
 /// history queries.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq, Eq, ValidateEq)]
 pub struct SystemMetadata {
     /// History of ingress messages as they traversed through the
     /// system.
```

Comment:
Same here. Should we skip ingress history and/or streams?

Reply:
Streams are limited to 10MB per subnet, so I think they should be fine.

For the ingress history it is at most 4GiB so skipping may be justified.

Reply:
Could we have custom `ValidateEq` implementations? So that e.g. we do a full equality check of `IngressHistory` when smaller than `N` messages / `M` bytes and only a size and/or metadata comparison if larger.

Not asking that you do it in this MR, just whether it could/should be done in a follow-up (by you or someone else).

Reply:
Yes, just like for any trait (e.g. PartialEq) one can provide implementation w/o auto-deriving